client-js/client/client.ts

```diff
@@ -0,0 +1,707 @@
+  public get connected(): boolean {
+    return ["connected", "ready"].includes(this._transport.state);
+  }
+
+  public get transport(): Transport {
+    return this._transportWrapper.proxy;
+  }
+
+  public get state(): TransportState {
+    return this._transport.state;
+  }
+
+  public get version(): string {
+    return packageJson.version;
+  }
+
+  // ------ Device methods
+
+  public async getAllMics(): Promise<MediaDeviceInfo[]> {
+    return await this._transport.getAllMics();
+  }
+
+  public async getAllCams(): Promise<MediaDeviceInfo[]> {
+    return await this._transport.getAllCams();
+  }
+
+  public async getAllSpeakers(): Promise<MediaDeviceInfo[]> {
+    return await this._transport.getAllSpeakers();
+  }
+
+  public get selectedMic() {
+    return this._transport.selectedMic;
+  }
+
+  public get selectedCam() {
+    return this._transport.selectedCam;
+  }
+
+  public get selectedSpeaker() {
+    return this._transport.selectedSpeaker;
+  }
+
+  public updateMic(micId: string) {
+    this._transport.updateMic(micId);
+  }
+
+  public updateCam(camId: string) {
+    this._transport.updateCam(camId);
+  }
+
+  public updateSpeaker(speakerId: string) {
+    this._transport.updateSpeaker(speakerId);
+  }
+
+  public enableMic(enable: boolean) {
+    this._transport.enableMic(enable);
+  }
+
+  public get isMicEnabled(): boolean {
+    return this._transport.isMicEnabled;
+  }
+
+  public enableCam(enable: boolean) {
+    this._transport.enableCam(enable);
+  }
+
+  public get isCamEnabled(): boolean {
+    return this._transport.isCamEnabled;
+  }
+
+  public tracks(): Tracks {
+    return this._transport.tracks();
+  }
+
+  public enableScreenShare(enable: boolean) {
+    return this._transport.enableScreenShare(enable);
+  }
+
+  public get isSharingScreen(): boolean {
+    return this._transport.isSharingScreen;
+  }
+
+  // ------ Messages
+
+  /**
+   * Directly send a message to the bot via the transport.
+   * Do not await a response.
+   * @param msgType - a string representing the message type
+   * @param data - a dictionary of data to send with the message
+   */
+  @transportReady
+  public sendClientMessage(msgType: string, data?: unknown): void {
+    this._transport.sendMessage(
+      new RTVIMessage(RTVIMessageType.CLIENT_MESSAGE, {
+        t: msgType,
+        d: data,
+      } as ClientMessageData)
+    );
+  }
+
+  /**
+   * Directly send a message to the bot via the transport.
+   * Wait for and return the response.
+   * @param msgType - a string representing the message type
+   * @param data - a dictionary of data to send with the message
+   * @param timeout - optional timeout in milliseconds for the response
+   */
+  @transportReady
+  public async sendClientRequest(
+    msgType: string,
+    data: unknown,
+    timeout?: number
+  ): Promise<unknown> {
+    return new Promise((resolve, reject) => {
+      const msgData: ClientMessageData = { t: msgType, d: data };
+      this._messageDispatcher
```

Comment:
why not `await` syntax instead of `.then`? seems like you prefer `await` in `appendToContext()`